src/main/java/com/github/onsdigital/client/SessionClientImpl.java

```
@@ -0,0 +1,160 @@
+import java.io.InputStreamReader;
+
+import static java.text.MessageFormat.format;
+
+public class SessionClientImpl implements SessionClient {
+
+    private static Logger LOG = LoggerFactory.getLogger(SessionClientImpl.class);
+
+    private String host;
+    private String serviceAuthToken;
+    private Http http;
+
+    public SessionClientImpl(final String host, final String serviceAuthToken) {
+        this.host = host;
+        this.serviceAuthToken = serviceAuthToken;
+        this.http = new Http();
+    }
+
+    @Override
+    public SessionCreated createNewSession(String userEmail) throws SessionClientException {
+        SessionCreated sessionCreated = null;
+
+        if (StringUtils.isNotEmpty(userEmail)) {
+            sessionCreated = postSession(userEmail);
+        }
+
+        return sessionCreated;
+    }
+
+    private SessionCreated postSession(String userEmail) {
+        try {
+            String jsonStr = http.toJson(new CreateNewSession(userEmail));
+            HttpPost httpPost = http.createHttpPost(host, "/sessions", jsonStr);
+            httpPost.setHeader("Authorization", "Bearer " + serviceAuthToken);
+            return http.doPost(httpPost, createSessionResponseHandler());
+        } catch (Exception ex) {
+            throw new SessionClientException(ex);
+        }
+    }
+
+    ResponseHandler<SessionCreated> createSessionResponseHandler() {
+        return (response -> {
+            int status = response.getStatusLine().getStatusCode();
+
+            if (status != HttpStatus.SC_CREATED) {
+                String msg = format("create session returned incorrect status expected 201 but was {0}", status);
+                throw new SessionClientException(msg);
+            }
+
+            return getResponseEntity(response, SessionCreated.class);
+        });
+    }
+
+
+    @Override
+    public Session getSessionByID(String sessionID) throws SessionClientException {
+        ZebedeeSession session = null;
```

I think all of these method should be validating their inputs. If the null, empty or invalid we should throw an exception. It feels like it should be this services responsibility so let the caller know things weren't OK rather than the caller having to figure that out.